### PR TITLE
CNF-25114: Replace shell exec with Go stdlib where possible

### PR DESCRIPTION
**Pull request by sebrandon1**
## Summary

Replace shell command invocations (`RunBashInHostNamespace`) with Go standard library equivalents to eliminate shell injection surface area and improve testability.

- **seedcreator**: replace `cp` shell calls with `ops.CopyFile` for MCO config and `.origin` file backups; replace `rpm-ostree status > file` redirect with `RunInHostNamespace` + `WriteFile`
- **ops**: replace `lsblk | jq` pipeline with `lsblk --json` + Go JSON parsing via new `getExtraPartitionPath` helper (consistent with existing `ListBlockDevices` pattern)
- **ops**: replace `mkdir -p` shell call with `ops.MkdirAll` in `SetupContainersFolderCommands`; restore `chattr +i` on `/mnt/` on failure
- **ops**: replace `crictl ps -q | xargs crictl stop` pipeline with Go goroutines using bounded concurrency (semaphore of 10, matching original `--max-procs`)

Partially addresses [#258](https://github.com/openshift-kni/lifecycle-agent/issues/258).

## Jira

[CNF-25114](https://issues.redhat.com/browse/CNF-25114)

## Test plan

- [x] `make build cli-build` passes
- [x] `make golangci-lint` passes
- [x] `make unittest` passes
- [x] New unit tests for `getExtraPartitionPath` (6 subtests + error case) and `StopClusterServices` (3 scenarios) cover JSON parsing edge cases and concurrent stop behavior

### CODE REVIEW

File: lca-cli/ops/ops.go [hunk 1]
```diff
@@ -735,10 +765,33 @@ func (o *ops) StopClusterServices() error {
...
+						errCh <- fmt.Errorf("failed to stop container %s: %w", containerID, stopErr)
+						return
+					}
+					errCh <- nil
+				}(id)
+			}
+			for range ids {
+				if stopErr := <-errCh; stopErr != nil {
+					return false, stopErr
+				}
```
[the remark this answers is not in the document]
Comment (sebrandon1) [reply lca-cli/ops/ops.go:894]: False positive - the error is intentionally discarded. This is a best-effort shutdown sequence: if graceful container stops fail, the code must proceed to stop CRI-O, which forcefully terminates any remaining containers. Propagating the error and returning early would leave CRI-O running, resulting in an incomplete shutdown - a worse outcome. The sandbox removal below follows the same pattern with an explicit "non-fatal" comment. The _ = assignment is deliberate.

**Comment by sebrandon1** [comment]
Rebased onto latest `add-context-to-execute-interface` (#6187) and resolved conflicts to propagate `context.Context` through the new shell-removal code paths.

`go test ./lca-cli/ops/... ./lca-cli/seedcreator/...` passes locally.

/retest-required

Ready for review after #6187 merges (or can merge together if preferred).

**Comment by sebrandon1** [comment]
Closing for the same reason as #6187 — stacked 59-file refactors aren't practical to backport. #258 stays open.
